gcp/client.go

```diff
@@ -56,12 +57,13 @@ func (c *DefaultClient) Deploy(ctx context.Context) error {
 	return nil
 }
 
-func New(ctx context.Context, projectName string, environment string) (*DefaultClient, error) {
-	m, err := deploymentmanager.NewService(ctx)
+func New(ctx context.Context, projectName string, environment string, credentials string) (*DefaultClient, error) {
+	auth := option.WithCredentialsFile(credentials)
```

Comment:
I think you only need this if credentials != ""

Reply:
i think credentials never will be empty, because its coming from the variable `GOOGLE_APPLICATION_CREDENTIALS` as default, and if the user dont inform the client will back an error. Still im throwing an error now case is empty

Reply:
More specific errors are better certainly.

Reply:
@gabrielbussolo I don't think you need to throw an error at this stage, as credentials are not mandatory.

```go
var authOpts []option.ClientOption
...
if credentials != "" {
    authOpts = append(authOpts, option.WithCredentialsFile(credentials))
}

deploymentmanager.NewService(ctx, authOpts...)
```

Reply:
i dindt undeerstand why credentials are “mandatory” if we cant set up the client without it